### ISSUE TITLE
Syntax Highlighting lost when passing Enum as Parameter

### ISSUE THREAD

**Issue by PirminStraub** [Galarius/vscode-opencl#11]
Hi Galarius, I just stumbled over the fact that the Syntax Highlighting is lost when passing an Enum as Parameter. Here is an example:
```
enum myEnum
{
	enumEntryOne, 
	enumEntryTwo
};

// float myFunction(float fValue, int enumValue) // syntax highlighting works in myFunction
float myFunction(float fValue, enum myEnum enumValue) // syntax highlighting works LOST in myFunction
{
	float returnValue;
	switch (enumValue)
	{
		case enumEntryOne:
			returnValue = 1.0f * fValue;
			break;
		case enumEntryTwo:
			returnValue = 2.0f * fValue;
			break;
	}
	return returnValue;
}
```
Keep up the good work! It is much appreciated!

**Comment by Galarius** [comment]
Hi, @PirminStraub! 

1. Which Color Theme you are using in VS Code?
2. What is the version of VS Code?

**Comment by PirminStraub** [comment]
1. Dark+ (default dark) - though the issue persists throughout all the other Color Themes as far as I could make out.
2. Version 1.35.1- seems to be the current one (for Windows)

**Comment by PirminStraub** [comment]
![vscode-syntaxhighlighting-1](https://user-images.githubusercontent.com/42964073/60292045-8c179300-991c-11e9-8957-93c8fd56623e.JPG)
![vscode-syntaxhighlighting-2](https://user-images.githubusercontent.com/42964073/60292066-9cc80900-991c-11e9-9b74-97f7c03dd04b.JPG)

**Comment by Galarius** [comment]
@PirminStraub, the `vscode-opencl` extension provides syntax highlighting for OpenCL keywords only. For C/C++ It is reusing grammar from [ms-vscode.cpptools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools). I have tried your sample in regular `.c` and `.cpp` files. In `.c` file the `enum` prefix was highlighted correctly, and in `.cpp` file the syntax highlighting was broken just like in your example. There are few workarounds I can suggest:

1. Fixing `C++` syntax highlighting by installing [Better Syntax](https://marketplace.visualstudio.com/items?itemName=jeff-hykin.better-cpp-syntax#review-details) extension. This extension will fix this and some other issues not only for regular `C++` code but also for `OpenCL`.

2. The other option is to modify your code in the following way (by adding `typedef`):
```c
typedef enum myEnum
{
	enumEntryOne, 
	enumEntryTwo
} myEnum;

float myFunction(float fValue, myEnum enumValue)
{ ... }
```

3. If you can switch to OpenCL C++ you won't need `enum` prefix before function argument's type.

If I find the way to extend or partially replace C++ highlighting rules without creating a full copy of [cpp.tmLanguage.json](https://raw.githubusercontent.com/microsoft/vscode/master/extensions/cpp/syntaxes/cpp.tmLanguage.json) I will implement it.

**Comment by Galarius** [comment]
Fixed in [0.6.1](https://github.com/Galarius/vscode-opencl/releases/tag/0.6.1)
